File: baekjoon/dp/1463-make-1.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int d[1000001];

int main() {
	int n;
	cin >> n;
	for(int i = 2; i <= n; i++) {
		int t = d[i - 1];
		if (i % 2 == 0) {
			t = min(t, d[i / 2]);
		}
		if (i % 3 == 0) {
			t = min(t, d[i / 3]);
		}
		d[i] = t + 1;
	}
	cout << d[n] << endl;
	return 0;
}

File: baekjoon/dp/1463-make-1.py
n = int(input())
d = [0] * 10000000
d[1] = 0
for i in range(2, n + 1):
    v = d[i - 1]
    if i % 3 == 0:
        v = min(d[i // 3], v)
    if i % 2 == 0 :
        v = min(d[i // 2], v)
    d[i] = v + 1

print(d[n])